src/java/org/apache/cassandra/cql3/functions/FunctionParameter.java

```diff
@@ -83,9 +86,9 @@ static FunctionParameter optional(FunctionParameter wrapped)
             public AbstractType<?> inferType(String keyspace,
                                              AssignmentTestable arg,
                                              @Nullable AbstractType<?> receiverType,
-                                             List<AbstractType<?>> previousTypes)
+                                             List<AbstractType<?>> inferredTypes)
```

Comment:
do we need a  annotation  of nullable for infferredType as line 60 ?

Reply:
We do, added